File: leetcode/two-sum.rs
use std::collections::HashMap;

impl Solution {
    pub fn two_sum(nums: Vec<i32>, target: i32) -> Vec<i32> {
        let mut hm: HashMap<i32,i32> = HashMap::new();
        hm.insert(nums[0],0);
        for i in 1..nums.len() {
            let c = nums[i];
            let req = target - c;
            if hm.contains_key(&req) {
                return vec![i as i32, *hm.get(&req).unwrap()];
            }
            hm.insert(c,i as i32);
        }
        return vec![];
    }
}

File: leetcode/TwoSum.java
package leetcode;

import java.util.HashMap;

class TwoSum {
    public int[] twoSum(int[] nums, int target) {
        HashMap<Integer, Integer> map = new HashMap<>();

        for (int i = 0; i < nums.length; i++) {
            int temp = target - nums[i];
            if (map.containsKey(nums[i])) {
                int[] res = { map.get(nums[i]), i };
                return res;
            }
            map.put(temp, i);
        }

        return null;
    }
}